lib/aws/request.ex

```diff
@@ -324,6 +324,15 @@ defmodule AWS.Request do
       nil ->
         {headers, params}
 
+      param_value when is_map(param_value) ->
+        headers =
+          Enum.reduce(param_value, headers, fn {k, v} ->
+            {header_name <> k, v}
```

Comment:
Can you clarify this step here? Perhaps show an example of a `param_value` map and the resulting `headers` as a reply to this comment. No need to add it as a comment in the code but I'm just a little unclear how concatenating the `k` with the `header_name` would produce something that AWS accepts? 🤔

Reply:
Considering that `build_params` is not private, consider adding a test for this as a means to document this now that I think about it 👍

Reply:
nvm, I see now 👍